src/apps/tezos/sign_tx.py

```diff
@@ -0,0 +1,155 @@
+import ustruct
+
+from trezor import wire
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.messages import TezosContractType
+from trezor.messages.TezosSignedTx import TezosSignedTx
+
+from apps.common import seed
+from apps.tezos.helpers import (
+    TEZOS_CURVE,
+    TEZOS_ORIGINATED_ADDRESS_PREFIX,
+    TEZOS_SIGNATURE_PREFIX,
+    b58cencode,
+    tezos_get_address_prefix,
+)
+from apps.tezos.layout import *
+
+
+async def sign_tx(ctx, msg):
+    address_n = msg.address_n or ()
+    node = await seed.derive_node(ctx, address_n, TEZOS_CURVE)
+
+    if msg.transaction is not None:
+        to = _get_address_from_contract(msg.transaction.destination)
+        await require_confirm_tx(ctx, to, msg.transaction.amount)
+        await require_confirm_fee(ctx, msg.transaction.amount, msg.transaction.fee)
+
+    elif msg.origination is not None:
+        source = _get_address_from_contract(msg.origination.source)
+        await require_confirm_origination(ctx, source)
+        await require_confirm_origination_fee(
+            ctx, msg.origination.balance, msg.origination.fee
+        )
+
+    elif msg.delegation is not None:
+        source = _get_address_from_contract(msg.delegation.source)
+
+        delegate = None
+        if msg.delegation.delegate is not None:
+            delegate = _get_address_by_tag(msg.delegation.delegate)
+
+        if delegate is not None and source != delegate:
+            await require_confirm_delegation_baker(ctx, delegate)
+            await require_confirm_set_delegate(ctx, msg.delegation.fee)
+        # if account registers itself as a delegate
+        else:
+            await require_confirm_register_delegate(ctx, source, msg.delegation.fee)
+
+    else:
+        raise wire.DataError("Invalid operation")
+
+    opbytes = _get_operation_bytes(msg)
```

Comment:
Have a look on Stellar, we usually declare a bytearray, which we then pass as an argument. I'm providing an example in one of the following comments.